.github/workflows/build.yml

```
@@ -21,15 +21,39 @@ jobs:
 
     strategy:
       matrix:
-        node-version: ['22', '24']
+        node-version:
+          - { v: 22 }
+          - { v: 24, isLatest: yes }
 
     steps:
     - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
-    - name: Use Node.js ${{ matrix.node-version }}
+    - name: Use Node.js ${{ matrix.node-version.v }}
       uses: actions/setup-node@49933ea5288caeca8642d1e84afbd3f7d6820020 # v4.4.0
       with:
-        node-version: ${{ matrix.node-version }}
+        node-version: ${{ matrix.node-version.v }}
     - run: npm clean-install
     - run: npm audit signatures
     - run: npm run all
     - run: npm run test
+
+    - name: '[PR Preview] Upload dist folder'
+      if: ${{github.event_name == 'pull_request' && matrix.node-version.isLatest}}
+      uses: actions/upload-artifact@v4
```

we should now stick to always pin the exact sha hash for the exact version we want to use, see #11103

```suggestion
      uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 # v4.6.2
```

(there is one more occurrence of this in this file and a couple more `uses` in `deploy-pr-preview.yml` )